### PR TITLE
#2892 Added a generator test case model in Dynawo Examples

### PR DESCRIPTION
**Pull request by joyelfeghali**
**Checklist before requesting a review**

*use `'[x]'` to check the checkboxes, or submit the PR and then click the checkboxes*

- [ ] unit tests and non-regression tests were added (new model, new feature and bug fix)
- [ ] main documentation was updated (update of input/output file, 3rd party, model, repository organization, solver)
- [ ] example documentations were updated (new example in examples folder)
- [ ] the corresponding milestone was added in the ticket and in this PR
- [ ] if this PR modifies the parameters or inputs/outputs of a model/solver: the corresponding xsl was added in util/xsl and platform DB were updated
- [ ] if this PR modifies a dictionary: the corresponding french dictionary was updated

### CODE REVIEW

**Comment by rosiereflo** [comment]
Please make sure to update the examples documentation when the testcase is ready :)

File: dynawo/sources/Models/Modelica/Dynawo/Examples/Generators/System1Model.mo [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
...e within [QMinPu, QMaxPu]. In the GeneratorPQProp model (used for Generator 1), QMinPu and QMaxPu are given. In the GeneratorPQPropdiagramPQ model (used for Generator 2), QMinPu and QMaxPu are calculated by referring to the PQ diagram while knowing the value of PGenPu.</div><div><br></div><div>At t = 50s, Line1bis is disconnected from the infinite bus; thus, the voltage at the generator level will increase. Signal NQ connected to both generators is calculated with a PI controller and will be different than 0 since URegulated is greater than URef. As a consequence, the reactive power generation QGenPu of the generators will decrease; thus, the URegulated will decrease following the PI control.</div><div>The freezing should be activated to stop the integration when the reactive power limits are reached.</div><div><br></div><div>At t = 70s, SignalN modeled as a step function will modify the generated active power PGenPu of generator 1 and generator 2 as long as it stays within the minimum and maximum limits. Here, PGenPu will be increased, causing the URegulated at the generator level to decrease below URef. Signal NQ will then increase the voltage.</div></div></div></body></html>"));
```
Comment (cossartque) [inline dynawo/sources/Models/Modelica/Dynawo/Examples/Generators/System1Model.mo:81]: Maybe we can first describe the test case : two generators, one infinite bus, two lines. The generators use a coordinated voltage control (SignalNQ) that dispatches the needed reactive power between the two generators to control the voltage at the generators connection bus. We simulate the disconnection of a line, bla bla.
We can also say here that SignalN is simply modelled by a step and not a proper regulation, as the infinite bus is here to balance the generation and consumption. Here N is simply used to model an active power reference step.

SignalNQ does not increase the voltage directly, it increases Q, which then results in an increase in the voltage.

File: dynawo/sources/Models/Modelica/Dynawo/Examples/Generators/System1Model_INIT.mo [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+*/
+
+model System1Model_INIT "Initialisation model for System 1"
+  extends AdditionalIcons.Init;
+
+  Electrical.Machines.SignalN.GeneratorPQProp_INIT Generator1(P0Pu = -1.5, PMax = 1000000, PMin = -1000000, Q0Pu = -0.5, QMax = 62, QMin = -200, U0Pu = 1.05, UPhase0 = 0) annotation(
+    Placement(visible = true, transformation(origin = {-28, 48}, extent = {{-10, -10}, {10, 10}}, rotation = 0)));
+  Electrical.Machines.SignalN.GeneratorPQPropDiagramPQ_INIT Generator2(P0Pu = -1.5, PMax = 1000000, PMin = -1000000, Q0Pu = -0.5, QMax0 = 200, QMin0 = -185, U0Pu = 1.05, UPhase0 = 0) annotation(
+    Placement(visible = true, transformation(origin = {-28, 16}, extent = {{-10, -10}, {10, 10}}, rotation = 0)));
+  annotation(
```
Comment (cossartque) [inline dynawo/sources/Models/Modelica/Dynawo/Examples/Generators/System1Model_INIT.mo:23]: One line before annotation and no line after

File: dynawo/sources/Models/Modelica/Dynawo/Examples/Generators/System1Model_INIT.mo [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+* This Source Code Form is subject to the terms of the Mozilla Public
+* License, v. 2.0. If a copy of the MPL was not distributed with this
+* file, you can obtain one at http://mozilla.org/MPL/2.0/.
+* SPDX-License-Identifier: MPL-2.0
+*
+* This file is part of Dynawo, an hybrid C++/Modelica open source suite
+* of simulation tools for power systems.
+*/
+
+model System1Model_INIT "Initialisation model for System 1"
```
Comment (cossartque) [inline dynawo/sources/Models/Modelica/Dynawo/Examples/Generators/System1Model_INIT.mo:16]: Initialisation model is enough (if we give up the System 1 name)

File: dynawo/sources/Models/Modelica/Dynawo/Examples/Generators/package.mo [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+* This Source Code Form is subject to the terms of the Mozilla Public
+* License, v. 2.0. If a copy of the MPL was not distributed with this
+* file, you can obtain one at http://mozilla.org/MPL/2.0/.
+* SPDX-License-Identifier: MPL-2.0
+*
+* This file is part of Dynawo, an hybrid C++/Modelica open source suite
+* of simulation tools for power systems.
+*/
+
+package Generators
```
Comment (cossartque) [inline dynawo/sources/Models/Modelica/Dynawo/Examples/Generators/package.mo:16]: To be renamed

File: dynawo/sources/Models/Modelica/Dynawo/Examples/Generators/package.order [hunk 5]
```diff
@@ -0,0 +1,2 @@
+System1Model
```
Comment (cossartque) [inline dynawo/sources/Models/Modelica/Dynawo/Examples/Generators/package.order:1]: To be renamed, thanks

**Review by cossartque — CHANGES_REQUESTED** [review]
A few remarks on the form mainly

File: dynawo/sources/Models/Modelica/Dynawo/Examples/DynaFlow/IllustrativeExamples/CoordinatedVControl.mo [hunk 6]
```diff
@@ -0,0 +1,83 @@
...
+* License, v. 2.0. If a copy of the MPL was not distributed with this
+* file, you can obtain one at http://mozilla.org/MPL/2.0/.
+* SPDX-License-Identifier: MPL-2.0
+*
+* This file is part of Dynawo, an hybrid C++/Modelica open source suite
+* of simulation tools for power systems.
+*/
+
+model CoordinatedVControl "System with two generators and a coordinated voltage control"
+  import Modelica.ComplexMath;
```
Comment (cossartque) [inline dynawo/sources/Models/Modelica/Dynawo/Examples/DynaFlow/IllustrativeExamples/CoordinatedVControl.mo:17]: This line can be deleted

File: dynawo/sources/Models/Modelica/Dynawo/Examples/DynaFlow/IllustrativeExamples/CoordinatedVControl.mo [hunk 7]
```diff
@@ -0,0 +1,83 @@
...
+  import Dynawo;
+  import Dynawo.Electrical;
```
Comment (cossartque) [inline dynawo/sources/Models/Modelica/Dynawo/Examples/DynaFlow/IllustrativeExamples/CoordinatedVControl.mo:19]: This line can be deleted

File: dynawo/sources/Models/Modelica/Dynawo/Examples/DynaFlow/IllustrativeExamples/CoordinatedVControl.mo [hunk 8]
```diff
@@ -0,0 +1,83 @@
...
+  Dynawo.Electrical.Buses.InfiniteBus infiniteBus1(UPhase = 0, UPu = 1.0305614) annotation(
+    Placement(visible = true, transformation(origin = {40, 40}, extent = {{-10, -10}, {10, 10}}, rotation = 0)));
+  Electrical.Controls.Basics.Step SignalN(Height = 0.1, Value0 = 0, tStep = 70) annotation(
+    Placement(visible = true, transformation(origin = {-90, 10}, extent = {{-10, -10}, {10, 10}}, rotation = 0)));
+  Dynawo.Electrical.Controls.Voltage.VRRemote SignalNQ(FreezingActivated = true, Gain = 1, NbGenMax = 2, U0 = 1.05, URef0 = 1.05, tIntegral = 10) annotation(
+    Placement(visible = true, transformation(origin = {-90, 64}, extent = {{-10, -10}, {10, 10}}, rotation = 0)));
+  Dynawo.Electrical.Events.Event.SingleBooleanEvent DiconnectLine(stateEvent1 = true, tEvent = 50)  annotation(
+   Placement(visible = true, transformation(origin = {-90, 36}, extent = {{-10, -10}, {10, 10}}, rotation = 0)));
+
+equation
```
Comment (cossartque) [inline dynawo/sources/Models/Modelica/Dynawo/Examples/DynaFlow/IllustrativeExamples/CoordinatedVControl.mo:41]: You can organize in sections here too.

File: dynawo/sources/Models/Modelica/Dynawo/Examples/DynaFlow/IllustrativeExamples/CoordinatedVControl_INIT.mo [hunk 9]
```diff
@@ -0,0 +1,26 @@
...
+* License, v. 2.0. If a copy of the MPL was not distributed with this
+* file, you can obtain one at http://mozilla.org/MPL/2.0/.
+* SPDX-License-Identifier: MPL-2.0
+*
+* This file is part of Dynawo, an hybrid C++/Modelica open source suite
+* of simulation tools for power systems.
+*/
+
+model CoordinatedVControl_INIT "Initialisation model"
+  extends AdditionalIcons.Init;
```
Comment (cossartque) [inline dynawo/sources/Models/Modelica/Dynawo/Examples/DynaFlow/IllustrativeExamples/CoordinatedVControl_INIT.mo:17]: import Dynawo needed

File: dynawo/sources/Models/Modelica/Dynawo/Examples/DynaFlow/IllustrativeExamples/CoordinatedVControl_INIT.mo [hunk 10]
```diff
@@ -0,0 +1,26 @@
...
+
+  Electrical.Machines.SignalN.GeneratorPQProp_INIT Generator1(P0Pu = -1.5, PMax = 1000000, PMin = -1000000, Q0Pu = -0.5, QMax = 62, QMin = -200, U0Pu = 1.05, UPhase0 = 0) annotation(
+    Placement(visible = true, transformation(origin = {-28, 48}, extent = {{-10, -10}, {10, 10}}, rotation = 0)));
+  Electrical.Machines.SignalN.GeneratorPQPropDiagramPQ_INIT Generator2(P0Pu = -1.5, PMax = 1000000, PMin = -1000000, Q0Pu = -0.5, QMax0 = 200, QMin0 = -185, U0Pu = 1.05, UPhase0 = 0) annotation(
```
Comment (cossartque) [inline dynawo/sources/Models/Modelica/Dynawo/Examples/DynaFlow/IllustrativeExamples/CoordinatedVControl_INIT.mo:21]: Dynawo.Electrical

**Review by cossartque — CHANGES_REQUESTED** [review]

**Review by cossartque — APPROVED** [review]

**Pull request merged**